src/chunk_constraint.c

```diff
@@ -640,3 +678,72 @@ chunk_constraint_recreate(ChunkConstraint *cc, Oid chunk_oid)
 	performDeletion(&constrobj, DROP_RESTRICT, 0);
 	chunk_constraint_create_on_table(cc, chunk_oid);
 }
+
+static void
+chunk_constraint_rename_on_chunk_table(int32 chunk_id, char *old_name, char *new_name)
+{
+	Chunk	   *chunk = chunk_get_by_id(chunk_id, 0, true);
+	RenameStmt	rename = {
+		.renameType = OBJECT_TABCONSTRAINT,
+		.relation = makeRangeVar(NameStr(chunk->fd.schema_name), NameStr(chunk->fd.table_name), 0),
+		.subname = old_name,
+		.newname = new_name,
+	};
+
+	RenameConstraint(&rename);
+}
+
+static bool
+chunk_constraint_rename_hypertable_tuple(TupleInfo *ti, void *data)
+{
+	RenameHypertableConstraintInfo *info = data;
+
+	bool		nulls[Natts_chunk_constraint];
+	Datum		values[Natts_chunk_constraint];
+	bool		repl[Natts_chunk_constraint] = {false};
+
+	HeapTuple	tuple;
+	NameData	new_hypertable_constraint_name;
+	NameData	new_chunk_constraint_name;
+	Name		old_chunk_constraint_name;
+
+	heap_deform_tuple(ti->tuple, ti->desc, values, nulls);
+
+	namestrcpy(&new_hypertable_constraint_name, info->newname);
+	chunk_constraint_choose_name(&new_chunk_constraint_name,
+								 false,
+								 0,
+								 info->newname,
+				  DatumGetInt32(values[Anum_chunk_constraint_chunk_id - 1]));
+
+
+	values[Anum_chunk_constraint_hypertable_constraint_name - 1] = NameGetDatum(&new_hypertable_constraint_name);
+	repl[Anum_chunk_constraint_hypertable_constraint_name - 1] = true;
+	old_chunk_constraint_name = DatumGetName(values[Anum_chunk_constraint_constraint_name - 1]);
+	values[Anum_chunk_constraint_constraint_name - 1] = NameGetDatum(&new_chunk_constraint_name);
+	repl[Anum_chunk_constraint_constraint_name - 1] = true;
+
+	chunk_constraint_rename_on_chunk_table(info->chunk_id, NameStr(*old_chunk_constraint_name), NameStr(new_chunk_constraint_name));
+
+	tuple = heap_modify_tuple(ti->tuple, ti->desc, values, nulls, repl);
+	catalog_update(ti->scanrel, tuple);
+	heap_freetuple(tuple);
+
+	return true;
+}
+
+int
+chunk_constraint_rename_hypertable_constraint(int32 chunk_id, const char *oldname, const char *newname)
+{
+	RenameHypertableConstraintInfo info = {
+		.hypertable_constraint_name = oldname,
+		.chunk_id = chunk_id,
+		.newname = newname,
+	};
+
+	return chunk_constraint_scan_by_chunk_id_internal(chunk_id,
```

Comment:
Here you are passing in a `RenameHypertableConstraintInfo` as data but the filter function assumes a `HypertableConstraintOperation`. This is dangerous as one must guarantee that the structs' field orders are compatible. A slight change in one of the structs will break this. I suggest this is better handled by either (1) struct embedding to ensure compatibility, or (2) making the filter and tuple handling function use the same struct.

Reply:
fixed